packer/ansible/roles/lvm-init/tasks/main.yml

```diff
@@ -0,0 +1,69 @@
+---
+- name: Wait until disk will be presend
+  wait_for:
+    path: /dev/xvdb
+
+- name: Create Volume Group
+  lvg:
+    vg: DataVG
+    pvs: "/dev/xvdb"
+
+- name: Create Thin Pool
+  register: thin_pool
+  failed_when: "thin_pool is failed and 'Sorry, no shrinking of DataLV to 0 permitted' not in thin_pool.msg"
+  lvol:
+    lv: DataLV
+    vg: DataVG
+    size: 100%FREE
+    opts: --thinpool ThinPool -V 1G
+
+- name: Format LVM
+  filesystem:
+    fstype: xfs
+    dev: /dev/DataVG/DataLV
+    opts: -L DATA
+
+- name: Mount
+  mount:
+    name: "/var/run/docker"
+    src: LABEL=DATA
+    fstype: xfs
+    opts: defaults,nofail
+    state: mounted
+
+- name: stat /boot/grub2/grub.cfg
+  stat: path=/boot/grub2/grub.cfg
+  register: grub2_conf
+
+- name: System Log                 | change /etc/default/grub
+  when: grub2_conf.stat.exists
+  replace:
+    dest: /etc/default/grub
+    regexp: 'rhgb'
+    replace: 'console=ttyS0,115200n8 earlyprintk=ttyS0,115200 rootdelay=300 xen_emul_unplug=unnecessary'
+  register: grub2_var
+
+- name: System Log                 | run grub2-mkconfig
+  when: grub2_var.changed and grub2_conf.stat.exists
+  command: grub2-mkconfig -o /boot/grub2/grub.cfg
+
+- name: Create dirs                | Create dirs
+  file: path={{ item }} state=directory
+  with_items:
+    - /var/lib/cloud/scripts/per-once
+    - /var/lib/cloud/scripts/per-boot
+
+- name: Data partition             | Auto resize LVM
+  template:
+    src: resize-xfs-lvm
+    dest: /var/lib/cloud/scripts/per-boot/resize-xfs
+    mode: 0755
+
+- name: Cron tasks                 | Add resize task to cron
+  cron:
+    name: "resize data partition"
+    minute: "*/5"
```

Comment:
Won't this frequency hurt the performance?

Reply:
We already have it for current OVA/AMI images. It minor actions and shouldn't affect performance